core/app_streaks_notifications.go

```diff
@@ -0,0 +1,299 @@
+
+	// for every config in course_configs:
+	// 		retrieve usercourse that passes 12am local time (how?) (also what if china to usa, experiencing same day mightnight twice)
+	//		if CompletedTasks? (refer to comments above):
+	//			streaks + 1; pause + 1
+	//			if usercourse's new pause > max_pause:
+	//				set to max_pause
+	//		else (incomplete task):
+	//			pause - 1
+	//			if pause < 0:
+	//				streak = 0; pause = 0
+	//		updateUserCourses([usercourses struct]) (why db usercourse doesn't have timezone?)
+	//		reset CompletedTasks?
+	// remember to assign initial pause when creating usercourse struct
+
+	for _, config := range courseConfigs {
+		tzOffsets := make(model.TZOffsets, 0)
+		var userCourses []model.UserCourse
+		// 0 for streak processtime
+		offset := 0 - nowSeconds
+		if config.StreaksNotificationsConfig.TimezoneName == "user" {
+			if offset >= minTZOffset && offset <= maxTZOffset {
+				tzOffsets = append(tzOffsets, offset)
+			}
+
+			if offset+utils.SecondsInDay <= maxTZOffset {
+				tzOffsets = append(tzOffsets, offset+utils.SecondsInDay)
+			}
+			if offset-utils.SecondsInDay >= minTZOffset {
+				tzOffsets = append(tzOffsets, offset-utils.SecondsInDay)
+			}
+
+			// load user courses for this course based on timezone offsets
+			userCourses, err = n.storage.FindUserCourses(nil, config.AppID, config.OrgID, nil, []string{config.CourseKey}, nil, tzOffsets.GeneratePairs(config.StreaksNotificationsConfig.PreferEarly), nil)
+			if err != nil {
+				n.logger.Errorf("processStreaks -> error finding user courses for course key %s: %v", config.CourseKey, err)
+				continue
+			}
+		} else {
+			configOffset := config.StreaksNotificationsConfig.TimezoneOffset
+			if offset == configOffset || offset+utils.SecondsInDay == configOffset || offset-utils.SecondsInDay == configOffset {
+				// load all user courses for this course
+				userCourses, err = n.storage.FindUserCourses(nil, config.AppID, config.OrgID, nil, []string{config.CourseKey}, nil, nil, nil)
+				if err != nil {
+					n.logger.Errorf("processStreaks -> error finding user courses for course key %s: %v", config.CourseKey, err)
+					continue
+				}
+			}
+		}
+
+		for i, userCourse := range userCourses {
+			if userCourse.CompletedTasks != nil {
+				uY, uM, uD := userCourse.CompletedTasks.Date()
+				y, m, d := now.Date()
+				pastY, pastM, pastD := now.AddDate(0, 0, -1).Date()
+				// edge case between time zone traveling
+				if uY == y && uM == m && uD == d {
+					continue
+					// yesterday
+				} else if uY == pastY && uM == pastM && uD == pastD {
+					userCourse.Streaks++
+					if userCourse.Streaks%config.PauseRewardStreak == 0 {
+						userCourse.Pauses++
+					}
+					if config.MaxPauses < userCourse.Pauses {
+						userCourse.Pauses = config.MaxPauses
+					}
+					// date earlier than yesterday, streak is broken
+				} else {
+					userCourse.Pauses--
+					if userCourse.Pauses < 0 {
+						userCourse.Streaks = 0
```

Comment:
The streak should always be set to 0 if we decrement the pauses, and let's only decrement pauses if it is greater than 0 to avoid setting `userCourse.Pauses` twice.

Reply:
"the streak should always be set to 0". So after if user has streaks=3, pauses=2 and skipped a day of work, then streaks=0,pauses=1?